prometheus-metrics-exposition-formats/src/test/java/io/prometheus/metrics/expositionformats/ExpositionFormatsTest.java

```diff
@@ -1825,24 +1810,24 @@ public void testUnknownMinimal() throws IOException {
     @Test
     public void testUnknownWithDots() throws IOException {
         String openMetrics = "" +
-                "# TYPE some_unknown_metric unknown\n" +
-                "# UNIT some_unknown_metric bytes\n" +
-                "# HELP some_unknown_metric help message\n" +
-                "some_unknown_metric{test_env=\"7\"} 0.7\n" +
+                "# TYPE some_unknown_metric_bytes unknown\n" +
+                "# UNIT some_unknown_metric_bytes bytes\n" +
```

Comment:
With Micrometer we ran into the same: we had verifications in our tests for the invalid output so I added an integration test with a Prometheus server: https://github.com/micrometer-metrics/micrometer/pull/5039

What do you think about doing something similar in the Prometheus Client?
The test starts an http server (could be the one in the Prometheus Client), starts a Prometheus sever, Prometheus scrapes the http server, the test verifies if the scraped metrics are available in Prometheus or not.

Reply:
YES, I actually had the same idea and started doing this for our latest module: `prometheus-metrics-exporter-pushgateway`.

https://github.com/prometheus/client_java/blob/main/integration-tests/it-pushgateway/src/test/java/io/prometheus/metrics/it/pushgateway/PushGatewayIT.java

I would like to implement similar tests for other modules over time.